cloud/services/compute/instances.go

```diff
@@ -102,15 +102,18 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*compute.Instance,
 				},
 			},
 		},
-		ServiceAccounts: []*compute.ServiceAccount{
+	}
+	if len(scope.GCPMachine.Spec.ServiceAccounts) > 0 {
+		input.ServiceAccounts = scope.GCPMachine.Spec.ServiceAccounts
+	} else {
```

Comment:
I'd remove the need for the else by keeping the defaults in the struct (like it was before), and overwriting it if there is one that's specified.